source/common/http/http2/codec_impl.cc

```diff
@@ -164,6 +164,9 @@ void ConnectionImpl::ClientStreamImpl::encodeHeaders(const RequestHeaderMap& hea
   } else if (headers.Method() && headers.Method()->value() == "CONNECT") {
     modified_headers = createHeaderMap<RequestHeaderMapImpl>(headers);
     modified_headers->setProtocol(Headers::get().ProtocolValues.Bytestream);
+    if (!headers.Path()) {
+      modified_headers->setPath("/");
+    }
```

Comment:
Is this required by the spec for H2? If so can you add comments with RFC reference?